async-stream/src/async_stream.rs

```diff
@@ -40,8 +40,8 @@ where
     type Item = T;
 
     fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
-        unsafe {
-            let me = Pin::get_unchecked_mut(self);
+        
+            let me = unsafe { Pin::get_unchecked_mut(self) };
```

Comment:
I would prefer to completely remove the unsafe code in this method by using pin-project-lite.